### PR TITLE
[Claim - Approve] -> USDC

### LINKED ISSUE #2151: PR-2130: 'GNO approved' message is displayed when TX is pending
Related to #2130

1. Send 'Approve GNO' token transaction to the wallet
2. Confirm transaction

**AR**: 'GNO approved' message is displayed when TX is still pending (and may finish with failure, btw)

**ER**: show something like a pending state as we do on the Swap form until TX is finished .

### PR DESCRIPTION
**Pull request by W3stside**
## Summary
Part of #2088 and #2087 and #2089 and closes #2151 

image

## Testing
1. needs acct with USDC and preferably GNO
2. approve USDC

### CODE REVIEW

File: src/custom/constants/tokens/index.ts
Comment (W3stside) [inline src/custom/constants/tokens/index.ts:83]: called `USDC_BY_CHAIN` because `USDC` is a native const used across the app in uni code.

**Comment by W3stside** [comment]
@anxolin @nenadV91 please review after merge

File: src/custom/pages/Claim/index.tsx
Comment (alfetopito) [inline src/custom/pages/Claim/index.tsx:260]: This is fine for now, but I'd rather have the approval logic inside the claim component.
We don't need to have this at the index, or do we?

Also, I still think we can do a generic approval flow rather than one for usdc and one for gno always even when they are not used.

Refactor for another day I guess

**Review by alfetopito — APPROVED** [review]
USDC approval working nicely

File: src/custom/pages/Claim/index.tsx
Comment (W3stside) [reply src/custom/pages/Claim/index.tsx:260]: probably easier, just didnt like "mapping" components having this logic

**Review by anxolin — COMMENTED** [review]
Only some small detail, can we just change a bit the description so we say `Approve investing vCOW using USDC`  for example?

image

File: src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx
Comment (W3stside) [reply src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx:94]: this is fixed in #2155

Comment (W3stside) [reply src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx:94]: > Only some small detail, can we just change a bit the description so we say `Approve investing vCOW using USDC` for example?
> 
> image

done: 05205fb

**Comment by W3stside** [comment]
> Only some small detail, can we just change a bit the description so we say `Approve investing vCOW using USDC` for example?
> 
> image

done b7cea63ca7cf90daeb035461e8efc456d5cd9555

**Comment by elena-zh** [comment]
LGTM!

**Pull request merged**
